tools/codegen/codespec/merge_attributes.go

```diff
@@ -1,57 +1,103 @@
 package codespec
 
-import "sort"
+import (
+	"sort"
+)
 
-// TODO: update to infer computability of inner nested attributes
-func mergeAttributes(pathParams, createRequest, createResponse, readResponse Attributes) Attributes {
-	merged := make(map[string]*Attribute)
-
-	addOrUpdate := func(attr *Attribute, computability ComputedOptionalRequired) {
-		if existingAttr, exists := merged[attr.Name.SnakeCase()]; exists {
-			if existingAttr.Description == nil || *existingAttr.Description == "" {
-				existingAttr.Description = attr.Description
-			}
+func mergeNestedAttributes(existingAttrs *Attributes, newAttrs Attributes, computability ComputedOptionalRequired, isFromResponse bool) {
```

Comment:
After debugging over the code I was surprised to see the logic for handling nested attributes is within `addOrUpdate`, and not during `mergeAttributes` where you can obtain for each nested attribute what value it has for each operation request/responses. Curious if you thought of this alternative or found limitation for this approach.

Reply:
Not sure I understand completely. `mergeAttributes` is intended as an entry point. 

>where you can obtain for each nested attribute what value it has for each operation request/responses.
can you elaborate how that may be better?

Reply:
I see. So each time an attribute is merged with addOrUpdate all nested attributes are being processed. Leaving some comments on `mergeAttributes`, `addOrUpdate`, and `mergeNestedAttributes` I believe will be very helpful for future reference.

Reply:
Right. We have to process nested attributes on every merge because create/read responses may contain additional attributes under the same parent attribute name that was initially included from the request.